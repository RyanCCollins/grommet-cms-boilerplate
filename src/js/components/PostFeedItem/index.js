/* @flow */
import React from 'react';
import Box from 'grommet/components/Box';
import Anchor from 'grommet/components/Anchor';
import Heading from 'grommet/components/Heading';
import Section from 'grommet/components/Section';
// $FlowFixMe grommet-cms required module not found
import { formatPrettyDate } from 'grommet-cms/utils';

export default function PostFeedItem(props: {
  post: {
    title: string,
    image: Object,
    slug: string,
    createdAt: string
  },
  colorIndex: string,
  postPath: string,
}) {
  const { post, postPath, colorIndex } = props;
  console.log('image', post.image.path, post);
  return (
    <Anchor className="post-feed-item--anchor" href={`${postPath}${post.slug}`}>
      <Section
        full
        colorIndex={colorIndex}
        texture={post.image.path}
        pad="large"
      >
        <Box
          align="center"
          justify="start"
          pad={{ vertical: 'large' }}
          full
        >
          <Heading strong align="center">
            {post.title}
          </Heading>
          <Heading align="center" tag="h3">
            {`Posted on ${formatPrettyDate(post.createdAt)}`}
          </Heading>
        </Box>
      </Section>
    </Anchor>
  );
}
