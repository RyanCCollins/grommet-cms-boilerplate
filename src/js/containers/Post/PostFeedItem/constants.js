/* @flow */
import type {
  LOAD_DATA_INITIATION_TYPE,
  LOAD_DATA_SUCCESS_TYPE,
  LOAD_DATA_FAILURE_TYPE
} from './flowTypes';
export const LOAD_DATA_INITIATION: LOAD_DATA_INITIATION_TYPE = 'POSTFEEDITEM/LOAD_DATA_INITIATION';
export const LOAD_DATA_SUCCESS: LOAD_DATA_SUCCESS_TYPE = 'POSTFEEDITEM/LOAD_DATA_SUCCESS';
export const LOAD_DATA_FAILURE: LOAD_DATA_FAILURE_TYPE = 'POSTFEEDITEM/LOAD_DATA_FAILURE';
