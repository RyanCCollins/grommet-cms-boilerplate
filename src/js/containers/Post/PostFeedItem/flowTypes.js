/* @flow */
export type LOAD_DATA_INITIATION_TYPE = 'POSTFEEDITEM/LOAD_DATA_INITIATION';
export type LOAD_DATA_SUCCESS_TYPE = 'POSTFEEDITEM/LOAD_DATA_SUCCESS';
export type LOAD_DATA_FAILURE_TYPE = 'POSTFEEDITEM/LOAD_DATA_FAILURE';

export type PostFeedItemState = {
  isLoading: boolean,
  loadingError?: { message: string }
}

export type PostFeedItemAction = {
  type: LOAD_DATA_INITIATION_TYPE | LOAD_DATA_SUCCESS_TYPE | LOAD_DATA_FAILURE_TYPE,
  loadingError?: { message: string }
}

export type PostFeedItemProps = {

}
