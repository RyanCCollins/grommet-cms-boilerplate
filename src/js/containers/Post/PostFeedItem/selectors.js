/* @flow */
import { createSelector } from 'reselect';

export const selectPostFeedItem = () => (state) => state.PostFeedItem;

export const selectMyProp = createSelector(
  selectPostFeedItem(),
  (PostFeedItem) => PostFeedItem.myProp
);
