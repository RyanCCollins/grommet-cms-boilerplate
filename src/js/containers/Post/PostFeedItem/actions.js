/* @flow */
import * as T from './constants';
import type { PostFeedItemAction } from './flowTypes';

export const loadDataInitiation = (): PostFeedItemAction => ({
  type: T.LOAD_DATA_INITIATION
});

export const loadDataSuccess = (): PostFeedItemAction => ({
  type: T.LOAD_DATA_SUCCESS
});

export const loadDataFailure = (error: { message: string }): PostFeedItemAction => ({
  type: T.LOAD_DATA_FAILURE,
  error
});
