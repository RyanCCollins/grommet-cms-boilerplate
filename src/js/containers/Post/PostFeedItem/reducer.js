/* @flow */
import * as T from './constants';
import type { PostFeedItemState, PostFeedItemAction } from './flowTypes';

export const initialState: PostFeedItemState = {
  isLoading: false,
  error: null
};

const postFeedItemReducer = (
  state: PostFeedItemState = initialState,
  action: PostFeedItemAction
): PostFeedItemState => {
  switch (action.type) {
    case T.LOAD_DATA_INITIATION:
      return {
        ...state,
        isLoading: true
      };
    case T.LOAD_DATA_SUCCESS:
      return {
        ...state,
        isLoading: false
      };
    case T.LOAD_DATA_FAILURE:
      return {
        ...state,
        isLoading: false,
        error: action.error
      };
    default:
      return state;
  }
};

export default postFeedItemReducer;
