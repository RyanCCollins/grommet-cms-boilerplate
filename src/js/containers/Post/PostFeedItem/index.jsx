/* @flow */
import React, { Component } from 'react';
import Box from 'grommet/components/Box';
import Card from 'grommet/components/Card';
import Paragraph from 'grommet/components/Paragraph';
import Section from 'grommet/components/Section';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as PostFeedItemActionCreators from './actions';
import { selectMyProp } from './selectors';
import type { PostFeedItemProps } from './flowTypes';

class PostFeedItem extends Component { // eslint-disable-line react/prefer-stateless-function
  props: PostFeedItemProps;
  render() {
    return (
      <Box>

      </Box>
    );
  }
}


const mapStateToProps = (state) => ({
  // myProp: selectMyProp(state)
});

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(
    PostFeedItemActionCreators,
    dispatch
  )
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PostFeedItem);
