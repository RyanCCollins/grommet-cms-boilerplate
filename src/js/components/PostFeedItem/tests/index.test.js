import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import PostFeedItem from '../index';

describe('<PostFeedItem />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <PostFeedItem />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
