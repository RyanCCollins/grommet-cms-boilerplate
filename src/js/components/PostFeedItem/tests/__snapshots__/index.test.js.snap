exports[`<PostFeedItem /> should render with default props 1`] = `
<Anchor
  className="post-feed-item--anchor"
  href="/post/watch-our-discover-video"
  method="push"
  tag="a">
  <Section
    colorIndex="grey-1-a"
    full={true}
    pad="large"
    texture="/uploads/media/2017/1/labs-blogfeed-1484268872525.gif">
    <Box
      align="center"
      announce={false}
      direction="column"
      focusable={true}
      full={true}
      justify="start"
      pad={
        Object {
          "vertical": "large",
        }
      }
      responsive={true}
      tag="div">
      <Heading
        align="center"
        strong={true}
        tag="h1">
        Watch our Discover Video
      </Heading>
      <Heading
        align="center"
        tag="h3">
        Posted on January 16, 2017
      </Heading>
    </Box>
  </Section>
</Anchor>
`;
